public/js/dashboard.js

```
@@ -1,96 +1,91 @@
-    },
-    mode: "same-origin",
-    method: "POST",
-    body: JSON.stringify(formBody),
-  });
+    const response = await fetch(`/user/${action}`, {
+        headers: {
+            "Content-Type": "application/json; charset=utf-8",
+        },
+        mode: "same-origin",
+        method: "POST",
+        body: JSON.stringify(formBody),
+    });
 
-  if (response.redirected) {
-    window.location = response.url;
-    return;
-  }
-  return await response.json();
+    if (response.redirected) {
+        window.location = response.url;
+        return;
+    }
+    return await response.json();
 }
 
 async function sendCommunicationOption(e) {
-  const { formAction, commOption, csrfToken } = e.target.dataset;
-  sendForm(formAction, { communicationOption: commOption, _csrf: csrfToken })
-    .then(data => {}) /*decide what to do with data */
-    .catch(e => {})/* decide how to handle errors */;
+    const { formAction, commOption, csrfToken } = e.target.dataset;
+    sendForm(formAction, { communicationOption: commOption, _csrf: csrfToken });
 }
 
 async function resendEmail(e) {
-  const resendEmailBtn = e.target;
-  const { formAction, csrfToken, emailId } = resendEmailBtn.dataset;
-  resendEmailBtn.classList.add("email-sent");
+    const resendEmailBtn = e.target;
+    const { formAction, csrfToken, emailId } = resendEmailBtn.dataset;
+    resendEmailBtn.classList.add("email-sent");
 
-  await sendForm(formAction, { _csrf: csrfToken, emailId })
-  .then(data => {
-    setTimeout( ()=> {
-      const span = resendEmailBtn.nextElementSibling;
-      span.classList.remove("hide");
-    }, 1000);
-  }) /*decide what to do with data */
-  .catch(e => {})/* decide how to handle errors */;
+    await sendForm(formAction, { _csrf: csrfToken, emailId })
+        .then(data => {
+            setTimeout( ()=> {
+                const span = resendEmailBtn.nextElementSibling;
+                span.classList.remove("hide");
+            }, 1000);
+        });
```

Same as above. Now errors aren't silently ignored.